### PR TITLE
a11y changes

### PR DESCRIPTION
**Pull request by 13zebras**
The first commit is for tab order. Some of what I did:

- rearrange the order of components on Host Room to make tab order work properly.
- modified modal from daisy-ui to make dialog behave in a fully accessible manner
- removed commented code.
- fixed responsive issues
- updated heading tags to be accessible
- removed tabindex

Biggest challenge:  radio buttons! I had to read considerably to figure them out, as well as experiment a ton. I think it is working as expected now.

Something I learned:  turning on Voiceover to hear/see what was being tabbed to. In some cases, it was hard to figure out where the tab was going. Voiceover literally told me!

### CODE REVIEW

File: src/components/HostControlButton.tsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ type Props = {
 	color: string
 	classes?: string
 	disabled?: boolean
+	// isDialogOpen: boolean
```
Comment (sandrina-p) [inline src/components/HostControlButton.tsx]: you can remove this right?

File: src/components/HostSettings.tsx [hunk 2]
```diff
@@ -124,7 +128,8 @@ export default function HostSettings({
 												/>
 												<label
 													htmlFor={storyPoint.toString()}
-													className='btn btn-primary btn-outline btn-checkbox-label'
+													className='btn btn-primary btn-outline btn-checkbox-label focus-visible:outline-indigo-300'
+													tabIndex={isDialogOpen ? 0 : -1}
```
Comment (sandrina-p) [inline src/components/HostSettings.tsx]: you can remove this tabindex, no need :)

File: src/components/NewUserDialog.tsx [hunk 3]
```diff
@@ -1,101 +1,136 @@
...
-							placeholder='Your First Name'
+							placeholder='Alexander'
 							name='userName'
 							className='input input-bordered input-primary h-10 max-w-72 w-full text-gray-200 placeholder:italic placeholder:text-primary/80 shadow-lg shadow-black/70'
-							aria-label='your first name input'
+							aria-label='your first name'
+							aria-describedby='first-name-error'
 							onChange={handleOnChange}
-							onKeyDown={handleKeyDown}
+							tabIndex={isDialogOpen ? 0 : -1}
```
Comment (sandrina-p) [inline src/components/NewUserDialog.tsx:112]: this tabindex can also be gone ;)

File: src/components/HostSettings.tsx (hunk 2 above)
Comment (sandrina-p) [reply src/components/HostSettings.tsx]: same for all the other tabindexes :)

File: src/components/HostSettings.tsx [hunk 4]
```diff
@@ -55,13 +57,14 @@ export default function HostSettings({
 				onClick={() => {
 					if (dialogRef.current) {
 						dialogRef.current.showModal()
+						setIsDialogOpen(true)
 					}
 				}}
-				className='btn btn-ghost size-6 min-h-6 p-0 border-0 hover:bg-transparent'
+				className='btn btn-ghost size-8 min-h-8 p-1 border-0 rounded-full hover:bg-transparent focus-visible:outline-none focus-visible:shadow-focusWhite'
 			>
```
Comment (sandrina-p) [inline src/components/HostSettings.tsx:65]: bug: Did you run Accessibility Insights For Web > "Automated checks" in every page/view? I don't think so, I run it and found a few errors! All related to buttons/icons ;)

Comment (13zebras) [reply src/components/HostSettings.tsx:65]: No I didn't!! But I will!

File: src/components/HostSettings.tsx (hunk 2 above)
Comment (13zebras) [reply src/components/HostSettings.tsx:132]: Already removed in the last commit

File: src/components/HostControlButton.tsx (hunk 1 above)
Comment (13zebras) [reply src/components/HostControlButton.tsx]: Yep, I simply missed that one

File: src/components/HostSettings.tsx (hunk 2 above)
Comment (13zebras) [reply src/components/HostSettings.tsx]: This tabindex actually represents something rather tricky. To style the checkbox to get the "button" effect, the LABEL is what has the styling while the input is actually display: hidden. It was much easier to make radio buttons look like a button, but not checkboxes. I can't remember the article I read that showed how to do this. 

So, without the tabindex, I can't get focus on the label. Is there some better way? You'll have to look at the CSS in devtools to really see what I mean.

Comment (13zebras) [reply src/components/HostSettings.tsx]: It should be simply tabindex={0} unless there is a better way.

File: src/components/HostSettings.tsx (hunk 4 above)
Comment (13zebras) [reply src/components/HostSettings.tsx:65]: I updated the buttons & svgs. I hope I did that right!

File: src/components/HostControlButton.tsx (hunk 1 above)
Comment (13zebras) [reply src/components/HostControlButton.tsx]: DONE

File: src/components/HostSettings.tsx (hunk 2 above)
Comment (13zebras) [reply src/components/HostSettings.tsx]: All other tabindex are removed. Let me know about this one and if there is a better way.

Comment (sandrina-p) [reply src/components/HostSettings.tsx]: checking 👀

File: src/components/AnimatedFishButton.tsx [hunk 5]
```diff
@@ -21,9 +21,13 @@ export default function AnimatedFishButton({
 			<button
 				type='button'
 				onClick={handleFishClick}
+				aria-label='AnimatedFish Button'
```
Comment (sandrina-p) [inline src/components/AnimatedFishButton.tsx]: todo: what did I tell you about saying "input" in a `<input>`? ;) The aria-label is only about the purposes (name), no need to say what it is (role). Check [WCAG name, role, value](https://www.getstark.co/wcag-explained/robust/compatible/name-role-value/) to better understand the difference among those 3 :)

File: src/components/HostSettings.tsx (hunk 4 above)
Comment (sandrina-p) [reply src/components/HostSettings.tsx:65]: Yes! My person preference is to add to all svg `aria-hidden`. 99% the times, the label goes to its parent (button, anchor, etc)

File: src/components/HostSettings.tsx (hunk 2 above)
Comment (sandrina-p) [reply src/components/HostSettings.tsx]: Ah! Okay, short lesson: you never ever set `display: none` to an input (or any other interactive element)! By doing that, you are hiding it from Assistive Technologies (like screen readers and keyboard). If you want to hide it visually, you find another way, for example, using the [`.sr-only` trick](https://www.a11yproject.com/posts/how-to-hide-content/). 
Check [inclusively hiding checkboxes](https://www.sarasoueidan.com/blog/inclusively-hiding-and-styling-checkboxes-and-radio-buttons/).

To know more about the different ways to add content: [hiding content responsibly](https://kittygiraudel.com/2021/02/17/hiding-content-responsibly/).

File: src/components/AnimatedFishButton.tsx (hunk 5 above)
Comment (13zebras) [reply src/components/AnimatedFishButton.tsx]: Ohhh.... I took you quite literally about "input" and didn't think "button, input, etc." Got it. I'll fix that.

File: src/components/HostSettings.tsx (hunk 2 above)
Comment (13zebras) [reply src/components/HostSettings.tsx]: Modified with Sara's method. It turned out to be the easiest method.

File: src/components/AnimatedFishButton.tsx (hunk 5 above)
Comment (13zebras) [reply src/components/AnimatedFishButton.tsx]: DONE

**Review by sandrina-p — APPROVED** [review]
A PR without `tabindex` 🎉

**Pull request merged**